api/models/tag.py

```diff
@@ -0,0 +1,58 @@
+from django.db import models
+import uuid
+from .team import Team
+from .ticket import Ticket
+from api.models.interfaces import HasUuid
+from hashlib import md5
+import re
+from django.core.validators import RegexValidator
+from django.utils.translation import gettext_lazy as _
+
+
+class Tag(HasUuid):
+    team_title_hash = models.CharField(max_length=256,
+                                       unique=True,
+                                       editable=False)
+    title = models.CharField(max_length=100,
+                             unique=False,
+                             validators=[
+                                 RegexValidator(re.compile(r'^[\w-]+$'),
+                                                _('Tag names must be word characters and dashes'),
+                                                'invalid'), ])
+    created = models.DateTimeField(auto_now_add=True)
+    activated = models.DateTimeField(null=True, blank=True)
+    deactivated = models.DateTimeField(null=True, blank=True)
+    team = models.ForeignKey(Team,
+                             on_delete=models.CASCADE,
+                             editable=True,
+                             null=False,
+                             related_name="tag")
+
+    class Meta:
+        ordering = ["id"]
+        unique_together = [["title", "team"]]
+        app_label = "api"
+
+    def __str__(self):
+        return f"Tag: {self.title}"
+
+    def _pre_create(self):
+        team = self.team
+        title = self.title
+
+        if not title:
+            raise ValueError("missing title")
+
+        if not team:
+            raise ValueError("missing team")
+
+        title_id = "{}".format(title)
+        team_id = "{}".format(team.id)
+        self.team_title_hash = md5(title_id.encode()).hexdigest() + md5(
+            team_id.encode()).hexdigest()
```

Comment:
this can be replaced with your new function right?

Reply:
Updated!